cmd/collect/work.go

```diff
@@ -2,24 +2,111 @@ package main
 
 import (
 	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
 
 	"github.com/GSA-TTS/jemison/internal/common"
 	"github.com/riverqueue/river"
+	"github.com/xeipuuv/gojsonschema"
 	"go.uber.org/zap"
 )
 
 //nolint:revive
 func (w *CollectWorker) Work(ctx context.Context, job *river.Job[common.CollectArgs]) error {
-	zap.L().Info("Collect has access to crawl record",
-		zap.String("scheme", job.Args.Scheme),
-		zap.String("host", job.Args.Host),
-		zap.String("path", job.Args.Path))
+	zap.L().Info("Collect Worker processing job", zap.String("data-json", job.Args.JSON))
 
-	if err := HandleBusinessLogic(job.Args); err != nil {
-		zap.L().Error("failed to handle business logic", zap.Error(err))
+	// Ensure schemas are initialized
+	if err := ensureSchemasInitialized(); err != nil {
+		return err
+	}
+
+	// Deserialize JSON data
+	rawData, err := deserializeJSON(job.Args.JSON)
+	if err != nil {
+		return err
+	}
+
+	// Check for `source` field and select schema
+	schema, err := selectSchema(rawData)
+	if err != nil {
+		return err
+	}
+
+	// Validate JSON and handle business logic
+	if err := validateAndHandleBusinessLogic(schema, job.Args); err != nil {
```

Comment:
I assume `handleBusinessLogic` will become `storeToS3` or similar. At that point, (again, assuming), we'll give it a better name? (A comment/question that doesn't particularly need a response for this commit.)